docs/reference/Configuration.md

```diff
@@ -0,0 +1,170 @@
+to arguments follows these rules:
+
+- All parameters are of the form `--parameterName=parameterValue`
+- `parameterName` is obtained by converting to lowercase the alphanumeric
+  characters in the parameter name (that is a free-form string) and replacing
+  non-alphanumeric characters with `-`.
+- If one or more sections contain the parameter, each section is treated as the
+  parameter name and joined using a `.`.
+
+So for example:
+
+```smalltalk
+MandatoryConfigurationParameter
+  named: 'Public URL'
+  describedBy: 'Public URL of the service'
+```
+
+needs to be provided in the command-line as:
+
+```bash
+launchpad start app --public-url=https://api.example.com
+```
+
+```smalltalk
+MandatoryConfigurationParameter
+  named: 'Port'
+  describedBy: 'Listening port'
+  inside: #( 'Communications' 'HTTP' )
+```
+
+needs to be provided in the command-line as:
+
+```bash
+launchpad start app --communications.http.port=8081
+```
+
+### Reading configuration from the OS environment
+
+When reading the configuration from the OS environment, parameter names conversion
+to environment variables follows these rules:
+
+- Convert parameter name to uppercase and replace non-alphanumeric characters
+  by `_`.
+- If one or more sections contain the parameter, each section is treated as the
+  parameter name and joined using a `__`.
+
+So for example:
+
+```smalltalk
+MandatoryConfigurationParameter
+  named: 'Public URL'
+  describedBy: 'Public URL of the service'
+```
+
+is converted into an environment variable named `PUBLIC_URL`.
+
+```smalltalk
+MandatoryConfigurationParameter
+  named: 'Port'
+  describedBy: 'Listening port'
+  inside: #( 'Communications' 'HTTP' )
+```
+
+is converted into an environment variable named `COMMUNICATIONS__HTTP__PORT`.
+
+### Reading configuration from setting files
+
+Configuration parameters can also be provided by reading a settings file.
+Supported formats are JSON and INI.
+
+When using the JSON format, parameters in the same section are nested in the file
+structure. Parameter and sections names are converter to JSON keys in lower
+camel-case format.
+
+For example:
+
+```smalltalk
+{
+  MandatoryConfigurationParameter
+    named: 'Port'
+    describedBy: 'Listening port'
+    inside: #( 'Communications' 'HTTP' ).
+  MandatoryConfigurationParameter
+    named: 'Transport Scheme'
+    describedBy: 'Transport Scheme (HTTP, HTTPS)'
+    inside: #( 'Communications' 'HTTP' ).
+}
+```
+
+needs to be provided as:
+
+```json
+{
+  "communications": {
+    "http": {
+      "port":8081,
+      "transportScheme":"https"
+    }
+  }
+}
+```
+
+When using the INI file format, sections cannot be arbitrarily nested.
+
+Section names to INI sections conversion follows these rules:
+
+- For each section name, replace non-alphanumeric characters with `-`.
+- If several sections contain the parameter, join each one using a `.`.
+
+Parameter names are converter to keys in lower camel-case format.
```

Comment:
Typo: converted

Reply:
I will fix it in future commits. Thanks.